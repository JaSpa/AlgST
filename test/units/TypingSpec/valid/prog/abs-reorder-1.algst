const_int : âˆ€(a:TU). Int -> a -> Int
const_int = \i [_:TU] _ -> i
