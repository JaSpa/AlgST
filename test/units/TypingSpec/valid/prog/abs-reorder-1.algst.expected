const_int#M.0
|
+- type
|  |
|  `- Type.Forall a#M.1:TU
|     |
|     `- Type.Arrow U
|        |
|        +- Type.Type
|        |  |
|        |  `- TypeRef
|        |     |
|        |     `- Int
|        |
|        `- Type.Arrow U
|           |
|           +- Type.Var a#M.1
|           |  |
|           |  `- TU
|           |
|           `- Type.Type
|              |
|              `- TypeRef
|                 |
|                 `- Int
|
+- params
|
`- definition
   |
   `- Exp.Abs U i#M.2
      |
      +- Type.Type
      |  |
      |  `- TypeRef
      |     |
      |     `- Int
      |
      `- Exp.TypeAbs _#M.3:TU
         |
         `- Exp.Abs U _#M.4
            |
            +- Type.Var _#M.3
            |  |
            |  `- TU
            |
            `- Exp.Var i#M.2

